src/js/angular/repositories/ontop-repo.directive.js

```diff
@@ -461,19 +465,27 @@ function ontopRepoDirective($uibModal, RepositoriesRestService, toastr, Upload,
             return Promise.resolve();
         };
 
-        // =========================
-        // Initialize component data
-        // =========================
-
-        Object.values(OntopFileType)
-            .forEach((ontopFileType) => {
-                const ontopFileInfo = new OntopFileInfo(ontopFileType);
-                $scope.formData.settings.ontopFiles.push(ontopFileInfo);
-            });
+        const clearFormData = () => {
+            $scope.formData = {
+                connectionInformation: new OntopConnectionInformation(),
+                settings: {
+                    additionalProperties: '',
+                    ontopFiles: []
+                }
+            };
 
-        $scope.getOntopFileInfo(OntopFileType.OBDA).required = true;
+            Object.values(OntopFileType)
+                .forEach((ontopFileType) => {
+                    const ontopFileInfo = new OntopFileInfo(ontopFileType);
+                    $scope.formData.settings.ontopFiles.push(ontopFileInfo);
+                });
 
+            $scope.getOntopFileInfo(OntopFileType.OBDA).required = true;
```

Comment:
Is this something like default? Can the required flag become part of the OntopFileType